### DISCUSSION TITLE
How to setup maintenance tasks with a custom controller?

### DISCUSSION THREAD

**Discussion by CodeGradox** [Shopify/maintenance_tasks#646] [Q&A] [answered]
Hi!

I want to try out maintenance tasks, but before I proceed, I need to know how to overwrite the controller provided by the engine. I'd like to add authorization to only allow admins to access the maintenance page.

**Comment by etiennebarrie** [comment, marked as answer]
One thing that can be done is to wrap the route in a constraint, like it's usually done with Sidekiq for example: https://github.com/mperham/sidekiq/wiki/Monitoring#authentication

You can also directly access `MaintenanceTasks::ApplicationController` and use a `before_action` there if you want, e.g.:

```ruby
# config/initializers/maintenance_tasks.rb
Rails.application.config.to_prepare do
  MaintenanceTasks::ApplicationController.class_eval do
    include Authentication
    before_action :authenticate
  end
end
```

**Comment by CodeGradox** [comment]
I tested with a custom constraint class, and it works really well! Thanks for the help 🧡
